### PR TITLE
feat/add instance pkg

### LINKED ISSUE #27: [FEAT] Release cycle management
**Is your feature request related to a problem? Please describe.**

As for now `cuebe` is only able to create or patch resources. It cannot detect a resource deletion in the release. In order to fully  manage release lifecycle, we need a way to detect every drifts.  

**Describe the solution you'd like**

This solution should only rely on tools already present. So Go SDK and a Kubernetes cluster.
We should also avoid messing with the cluster state. The less object we have to create, the better.

**Additional context**

This is an open discussion before actually implementing a solution.

### PR DESCRIPTION
**Pull request by b4nst**
- fix: fix context
- chore: add utils
- feat: refactor manifest
- feat: implements instance concept
- refactor: refactor commands
- chore: remove unused code

Resolves #27

### CODE REVIEW

File: cmd/cuebe/cmd/export.go [hunk 1]
```diff
@@ -48,73 +55,93 @@ cuebe export -i main.enc.yaml
...
+	for _, arg := range args {
+		// TODO move that to a function in the context package
+		if !path.IsAbs(arg) {
+			cwd, err := os.Getwd()
+			if err != nil {
+				return opts, fmt.Errorf("could not get working directory: %w", err)
+			}
+			arg = path.Join(cwd, arg)
+		}
+		if err := opts.BuildContext.Add(afero.NewBasePathFs(afero.NewOsFs(), arg)); err != nil {
```
Comment (astraios) [inline cmd/cuebe/cmd/export.go:96]: Just so I understand correctly what's happening, `.Add()` will copy each context (passed as command args) to the `BuildContext` which is a `fs` or type `MemMapFs`. So Afero returns you an "InMemory" FS when instantiated with `bctxt.New()`, then .Add will copy each detected context in that memory space for future operation?

**Review by astraios — COMMENTED** [review]
Add a couple of questions, but that's just to educate me. Looks sane to me.

File: cmd/cuebe/flag/build.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+type BuildOpt struct {
+	// Expressions are the expressions to extract manifest from.
+	Expressions []string
+	// Tags are a list of key value used as CUE tags.
+	Tags []string
+}
+
+// AddBuild adds the necessary flags for a build aware command.
+func AddBuild(fs *pflag.FlagSet) {
+	fs.StringSliceP(expression, expressionShort, []string{}, expressionDesc)
```
[the remark this answers is not in the document]
Comment (b4nst) [reply cmd/cuebe/flag/build.go:29]: Yup, you can use multiple `-e` now

File: cmd/cuebe/cmd/apply.go [hunk 3]
```diff
@@ -65,119 +66,116 @@ cuebe apply --dry-run
...
 
-	// InjectFiles
-	i, err := cmd.Flags().GetStringSlice("inject")
+	// dry-run
+	dr, err := cmd.Flags().GetBool("dry-run")
 	if err != nil {
 		return nil, fmt.Errorf("Failed parsing args: %w", err)
 	}
-	opts.InjectFiles = i
+	opts.DryRun = dr
```
[the remark this answers is not in the document]
Comment (b4nst) [reply cmd/cuebe/cmd/apply.go:95]: That's an old copy pasta, you're totally right. And that works for all of them. Will fix

**Pull request merged**